packages/shared/lib/sdk/sync.ts

```diff
@@ -307,6 +339,173 @@ export class NangoAction {
+                        // some providers only send path+query params in the link so we can immediately assign those to the endpoint
+                        config.endpoint = nextPageUrl;
+                    } else {
+                        const url: URL = new URL(nextPageUrl);
+                        config.endpoint = url.pathname + url.search;
+                    }
+                    delete config.params;
+                }
+            }
+            case PaginationType.OFFSET: {
+                const offsetPagination: OffsetPagination = paginationConfig as OffsetPagination;
+                const offsetParameterName: string = offsetPagination.offset_parameter_name;
+                let offset: number = 0;
+
+                while (true) {
+                    updatedBodyOrParams[offsetParameterName] = `${offset}`;
+
+                    this.updateConfigBodyOrParams(passPaginationParamsInBody, config, updatedBodyOrParams);
+
+                    const response: AxiosResponse = await this.proxy(config);
+
+                    const responseData: T[] = paginationConfig.response_data_path
+                        ? this.getNestedField(response.data, paginationConfig.response_data_path)
+                        : response.data;
+                    if (!responseData.length) {
+                        return;
+                    }
+
+                    yield responseData;
+
+                    if (responseData.length < 1) {
+                        // Last page was empty so no need to fetch further
+                        return;
+                    }
+
+                    offset += responseData.length;
+                }
+            }
+            default:
+                throw Error(`'${paginationConfig.type} ' pagination is not supported. Please, make sure it's one of ${Object.values(PaginationType)}`);
+        }
+    }
+
+    private getNextPageUrlFromBodyOrHeaders(nextUrlPagination: NextUrlPagination, response: AxiosResponse<any, any>, paginationConfig: Pagination) {
+        if (nextUrlPagination.link_rel) {
+            const linkHeader = parseLinksHeader(response.headers['link']);
+            return linkHeader?.[nextUrlPagination.link_rel]?.url;
+        } else if (nextUrlPagination.link_body_parameter_path) {
+            return this.getNestedField(response.data, nextUrlPagination.link_body_parameter_path);
+        }
+
+        throw Error(`Either 'link_rel' or 'link_body_parameter_path' should be specified for '${paginationConfig.type}' pagination`);
+    }
+
+    private updateConfigBodyOrParams(passPaginationParamsInBody: boolean, config: ProxyConfiguration, updatedBodyOrParams: Record<string, string>) {
+        if (passPaginationParamsInBody) {
+            config.data = updatedBodyOrParams;
+        } else {
+            config.params = updatedBodyOrParams;
+        }
+    }
+
+    private getNestedField(object: any, path: string, defaultValue?: any): any {
```

Comment:
Isn't there a library for this? e.g. Lodash? 

Example of using Lodash for this (from a previous Nango [version](https://github.com/NangoHQ/nango-sync/blob/main/packages/worker/lib/services/external.service.ts)): 
```ts
// Cursor-based pagination (with cursor field in the metadata of the response).
            if (
                sync.paging_cursor_metadata_response_path != null &&
                _.get(res.data, sync.paging_cursor_metadata_response_path) &&
                results.length < maxNumberOfRecords
            ) {
                pageCursor = _.get(res.data, sync.paging_cursor_metadata_response_path);
                continue;
            }

            // Cursor-based pagination (with cursor field in the last object of the response).
            if (
                sync.paging_cursor_object_response_path != null &&
                _.get(newResults[newResults.length - 1], sync.paging_cursor_object_response_path) &&
                results.length < maxNumberOfRecords
            ) {
                pageCursor = _.get(newResults[newResults.length - 1], sync.paging_cursor_object_response_path);
                continue;
            }

            // URL-based pagination (with URL field in the metadata of the response).
            if (sync.paging_url_path != null && isValidHttpUrl(_.get(res.data, sync.paging_url_path)) && results.length < maxNumberOfRecords) {
                sync.url = _.get(res.data, sync.paging_url_path);
                continue;
            }

            // URL-based pagination (with URL field in the Link header).
            if (sync.paging_header_link_rel != null && res.headers['link'] != null) {
                let linkHeader = parseLinksHeader(res.headers['link']);

                if (linkHeader != null && sync.paging_header_link_rel in linkHeader && isValidHttpUrl(linkHeader[sync.paging_header_link_rel]['url'])) {
                    let nextPageUrl = linkHeader[sync.paging_header_link_rel]['url'];
                    sync.url = nextPageUrl;
                    continue;
                }
            }

            // Offset + limit pagination
            if (sync.paging_offset_request_path != null && sync.paging_limit_request_path != null && sync.paging_limit != null) {
                if (newResults.length > 0) {
                    offset += newResults.length;
                    continue;
                } else {
                    // The last page was empty, so we should not count it.
                    page -= 1;
                }
            }
```

Reply:
Yeah, I have a `TODO` inside this method to figure out how to use `lodash`. For some reason, I couldn't easily import so I created this helper method to move forward and get back to this later.